packages/engine/src/scene/classes/Clouds.ts

```diff
@@ -0,0 +1,210 @@
+import {
+  Mesh,
+  InstancedBufferGeometry,
+  PlaneBufferGeometry,
+  ShaderMaterial,
+  Vector3,
+  InstancedBufferAttribute,
+  Texture,
+  RawShaderMaterial,
+  UniformsUtils,
+  Fog,
+  Vector2,
+  Color
+} from 'three'
+import loadTexture from '../../editor/functions/loadTexture'
+import SimplexNoise from 'simplex-noise'
+
+const vertexShader = `
+precision highp float;
+
+attribute vec4 particles;
+attribute float particleAngle;
+
+varying vec2 vUV;
+
+void main(){
+  vUV = uv;
+
+  vec3 rotatedPosition = position;
+  rotatedPosition.x = cos( particleAngle ) * position.x - sin( particleAngle ) * position.y;
+  rotatedPosition.y = sin( particleAngle ) * position.x + cos( particleAngle ) * position.y;
+  float scale = particles.w;
+  rotatedPosition *= scale;
+
+  vec4 mvPosition = modelViewMatrix * vec4(particles.xyz, 1);
+  mvPosition.xyz += rotatedPosition;
+  gl_Position = projectionMatrix * mvPosition;
+}
+`
+const fragmentShader = `
+precision highp float;
+
+uniform sampler2D map;
+
+uniform vec3 fogColor;
+uniform float fogNear;
+uniform float fogFar;
+
+varying vec2 vUV;
+
+void main() {
+  float depth = gl_FragCoord.z / gl_FragCoord.w;
+  float fogFactor = smoothstep( fogNear, fogFar, depth );
+
+  gl_FragColor = texture2D(map,  vUV);
+  gl_FragColor.w *= pow( gl_FragCoord.z, 20.0 );
+	gl_FragColor = mix( gl_FragColor, vec4( fogColor, gl_FragColor.w ), fogFactor );
+}
+`
+
+export class Clouds extends Mesh {
+  worldScale: Vector3
+  dimensions: Vector3 // Number of particles (x,y,z)
+  noiseZoom: Vector3
+  noiseOffset: Vector3
+  spriteScaleRange: Vector2
+  fogRange: Vector2
+  fogColor: Color
+  noise: SimplexNoise
+
+  constructor(texture: Texture, configs: any) {
+    const planeGeometry = new PlaneBufferGeometry(1, 1, 1, 1)
+    const geometry = new InstancedBufferGeometry()
+    geometry.index = planeGeometry.index
+    geometry.attributes = planeGeometry.attributes
+
+    const fog = new Fog(0x4584b4, -100, 3000)
+
+    const material = new ShaderMaterial({
+      uniforms: UniformsUtils.merge([
+        {
+          map: { value: texture },
+          fogColor: { type: 'c', value: fog.color },
+          fogNear: { type: 'f', value: fog.near },
+          fogFar: { type: 'f', value: fog.far }
+        }
+      ]),
+      vertexShader,
+      fragmentShader,
+      transparent: true,
+      depthWrite: false
+    })
+
+    material.uniforms.map.value = texture
+    super(geometry, material)
+
+    this.frustumCulled = false
+    this.worldScale = new Vector3(1000, 150, 1000)
+    this.dimensions = new Vector3(8, 4, 8)
+    this.noiseZoom = new Vector3(7, 11, 7)
+    this.noiseOffset = new Vector3(0, 4000, 3137)
+    this.spriteScaleRange = new Vector2(50, 100) // Min/Max
+    this.fogColor = new Color(0x4584b4)
+    this.fogRange = new Vector2(-100, 3000)
+    this.noise = new SimplexNoise('seed')
+
+    if (configs) {
```

Comment:
this can be removed, all parameters should be set explicitly

Reply:
Moved the defaults to CloudNode.ts